### DISCUSSION TITLE
mesh networking

### DISCUSSION THREAD

**Discussion by yanghaoyu412726** [open-sdr/openwifi#247] [Show and tell]
Hello Dr. Jiao, can openwifi support mesh networking?

**Comment by hegs123** [comment]
I also have some puzzles based on openwifi mesh networking, please ask Dr. Jiao to answer, thank you

**Comment by JiaoXianjun** [comment]
We never look into and try mesh.

I guess (pure guess), you need to enable some mesh related kernel configuration, and re-compile the kernel.

We have detailed README about how to re-compile the kernel. Please do some homework on this if you want to do it.

**Reply by yanghaoyu412726** [reply]
thanks!

**Comment by hegs123** [comment]
Thanks for Dr. Jiao's reply. I have made some attempts and met some specific problems. I have sent you the email. You can have a look if you are interested.

**Comment by JiaoXianjun** [comment]
Why not post them here. More people could help you.

**Comment by hegs123** [comment]
The mesh configuration is as follows：
->iw dev sdr0 interface add mesh0 type mp
->iw dev mesh0  set channel 1
->iw dev mesh0 set meshid mymesh
->ifconfig mesh0 hw ether 00:15:16:10:20:66（The mac address cannot be the same as sdr0; otherwise, the system cannot be started）
->ifconfig mesh0  10.0.0.8
->iw dev mesh0 mesh join mymesh
    error:  SIOCSIFFLAGS: Device or resource busy
    View the dmegs display  :  mesh0: TSF : max clockdrift=0; too small to adjust

The above is about the configuration steps of mesh network, but mesh_id cannot be added to the grid, and the kernel configuration also exists. I do not know the specific problem, which expert can give you advice would be very grateful

**Reply by zhangmazi123321** [reply]
Configure sdr0 after turning it off

**Reply by zhangmazi123321** [reply]
Can you tell me how you implemented mp mode?

**Comment by JiaoXianjun** [comment]
Our latest release has started to use the very new raspberry PI OS and linux kernel. Maybe it offers better support to mesh.
Could you please try again?
